promising/promise.py

```
@@ -323,13 +434,27 @@ def get_pending_children(self) -> set["Promise[Any]"]:
             try:
                 # In `asyncio.tasks` it was `_all_tasks` instead of
                 # `self._children`
-                children = list(self._children)
+                children = list[Promise[Any]](self._children)
             except RuntimeError:
                 i += 1
                 if i > 1000:  # noqa: PLR2004 (magic-value-comparison)
```

Ruff (0.15.1)

[warning] Unused `noqa` directive (non-enabled: `PLR2004`)

Remove unused `noqa` directive

(RUF100)